src/GNDStk/Tree/src/decl.hpp

```diff
@@ -0,0 +1,29 @@
+
+// -----------------------------------------------------------------------------
+// Tree::decl()
+// -----------------------------------------------------------------------------
+
+// const
+const nodeType &decl() const
```

Comment:
I realized today why I don't like this `decl` node. It's because it is not in GNDS. This `decl` node is only valid in an XML file. It shouldn't be part of GNDS. We have to write it when we write to XML, but we don't need to keep it. We can create all of the data in this node when and how we decide to write it. We simply don't need to store it.

Reply:
It is just another node, and isn't special as far as the tree structure is concerned.

I've written elsewhere about why I keep the XML declaration node around, in the event that we've read something from an XML file, or that we wish to write to an XML file. For the other file types: if they prove not to have something similar to the declaration node, then any declaration node that might be in the tree structure, won't play a role in writing to the other type of file.

Trees and nodes are designed in such a way that if you don't care about anything in an XML declaration node, or if such a beast doesn't exist (because you're not dealing with XML), then you don't even need to know that it's there. (Or not there.) Its existence has no effect on the manner in which GNDS trees are constructed or queried.

Reply:
I have read what you have written about the node and I'm suggesting we don't need to keep it around. It has nothing to do with GNDS and only makes this code more complicated. We don't need to keep it around because we get to decide these parameters when we write the XML file.